rust/src/core.rs

```diff
@@ -188,6 +193,16 @@ impl FilesystemScriptPrep {
                 rootfs.atomic_write_with_perms(path, contents, mode)?;
             }
         }
+        if std::path::Path::new(OSTREE_BOOTED).exists() {
```

Comment:
I think we were talking about needing to detect the kernel layout here too and skip wrapping the kernel install in that case right?

Which is...definitely unfortunate and messy but we're trying to dig ourselves out of the tech debt here. I'd say we we basically add a little helper in `kernel_install.rs` that looks for `/usr/lib/kernel/install.conf` and checks if it contains `layout=ostree` or so

Reply:
latest commit tries to do this.